[BUG] KudoSerializerTest failed Maximum pool size exceeded (failed to allocate 20.000000 GiB)

**Describe the bug**
related to https://github.com/NVIDIA/spark-rapids-jni/blob/branch-25.10/src/test/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializerTest.java#L49

```
[2025-09-24T04:10:09.364Z] [INFO] Running com.nvidia.spark.rapids.jni.kudo.KudoSerializerTest
[2025-09-24T04:10:09.364Z] [INFO] 
[2025-09-24T04:10:09.364Z] [INFO] Results:
[2025-09-24T04:10:09.364Z] [INFO] 
[2025-09-24T04:10:09.364Z] [INFO] ------------------------------------------------------------------------
[2025-09-24T04:10:09.364Z] [INFO] BUILD FAILURE
[2025-09-24T04:10:09.364Z] [INFO] ------------------------------------------------------------------------
[2025-09-24T04:10:09.364Z] [INFO] Total time: 01:03 h
[2025-09-24T04:10:09.364Z] [INFO] Finished at: 2025-09-24T04:10:02Z
[2025-09-24T04:10:09.364Z] [INFO] ------------------------------------------------------------------------
[2025-09-24T04:10:09.364Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project spark-rapids-jni: There are test failures.
[2025-09-24T04:10:09.364Z] [ERROR] 
[2025-09-24T04:10:09.364Z] [ERROR] Please refer to /home/jenkins/agent/workspace/peixinl-spark-rapids-jni_nightly-dev/target/surefire-reports for the individual test results.
[2025-09-24T04:10:09.364Z] [ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[2025-09-24T04:10:09.364Z] [ERROR] There was an error in the forked process
[2025-09-24T04:10:09.364Z] [ERROR] Error occurred in starting fork, check output in log
[2025-09-24T04:10:09.364Z] [ERROR] Could not allocate native memory: std::bad_alloc: out_of_memory: RMM failure at:/home/jenkins/agent/workspace/peixinl-spark-rapids-jni_nightly-dev/target/libcudf-install/include/rmm/mr/device/pool_memory_resource.hpp:261: Maximum pool size exceeded (failed to allocate 20.000000 GiB): std::bad_alloc: out_of_memory: CUDA error (failed to allocate 21474836480 bytes) at: /home/jenkins/agent/workspace/peixinl-spark-rapids-jni_nightly-dev/target/libcudf-install/include/rmm/mr/device/cuda_memory_resource.hpp:62: cudaErrorMemoryAllocation out of memory
[2025-09-24T04:10:09.364Z] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[2025-09-24T04:10:09.365Z] [ERROR] Error occurred in starting fork, check output in log
[2025-09-24T04:10:09.365Z] [ERROR] Could not allocate native memory: std::bad_alloc: out_of_memory: RMM failure at:/home/jenkins/agent/workspace/peixinl-spark-rapids-jni_nightly-dev/target/libcudf-install/include/rmm/mr/device/pool_memory_resource.hpp:261: Maximum pool size exceeded (failed to allocate 20.000000 GiB): std::bad_alloc: out_of_memory: CUDA error (failed to allocate 21474836480 bytes) at: /home/jenkins/agent/workspace/peixinl-spark-rapids-jni_nightly-dev/target/libcudf-install/include/rmm/mr/device/cuda_memory_resource.hpp:62: cudaErrorMemoryAllocation out of memory
```

Previously, this case has not been run on T4 gpu (randomly assigned to different GPUs in the resource pool)

**Steps/Code to reproduce bug**
Run the case on a GPU with less than 20GB memory (like Tesla T4)
```
[2025-09-24T03:05:51.030Z] |=========================================+========================+======================|
[2025-09-24T03:05:51.030Z] |   0  Tesla T4                       Off |   00000000:01:00.0 Off |                    0 |
[2025-09-24T03:05:51.030Z] | N/A   33C    P8              9W /   70W |       0MiB /  15360MiB |      0%      Default |
[2025-09-24T03:05:51.030Z] |                                         |                        |                  N/A |
[2025-09-24T03:05:51.030Z] +-----------------------------------------+------------------------+----------------------+
```

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/src/test/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializerTest.java
+++ b/src/test/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializerTest.java
@@ -816,7 +816,7 @@ public class KudoSerializerTest extends CudfTestBase {
     List<ColumnVector> allCols = new ArrayList<>();
     List<ColumnVector> tableCols = new ArrayList<>();
 
-    final int nonNullCount = Integer.MAX_VALUE / 512 - 21; // to avoid OOM
+    final int nonNullCount = Integer.MAX_VALUE / 128 - 21; // to avoid OOM
     final int nullCount = 11; // to add nulls
     final int totalCount = nonNullCount + nullCount;
 

